platform/android/scripts/generate-style-code.js

```diff
@@ -197,7 +197,12 @@ global.defaultValueJava = function(property) {
       case 'array':
              switch (property.value) {
               case 'string':
-                return '[' + property['default'] + "]";
+              case 'enum':
+                if (property['default'] !== undefined) {
```

Comment:
This change has to be applied to `darwin/scripts/generate-style-code.js` and please run `make darwin-style-code` after. That's why the iOS and macOS tests fail in this branch.

Let me know if you have any questions.

Reply:
It's a bit more than this, pls see https://github.com/mapbox/mapbox-gl-native/pull/14184/commits/9bc4994715b527df56189ca55619b64e8ccbcc3f